src/check/parse.zig

```diff
@@ -0,0 +1,195 @@
+const std = @import("std");
+const tokenize = @import("tokenize.zig");
+
+pub const Region = struct {
+    start: usize,
+    end: usize,
+};
+
+pub const Node = struct {
+    tag: Tag,
+    data: Data,
+    region: Region,
+
+    pub const Tag = enum {
+        NodeType1,
+        NodeType2,
+        // Add more node types as needed
+    };
+
+    pub const Data = union(Tag) {
```

Comment:
You will probably want an untagged union, like this:

```suggestion
    pub const Data = union {
```

Point being that when this code is no longer a stub, you will not have 1:1 correspondence between Tag and Data. Data will be like, binary_operation, unary_operation, and other misc stuff. The Tag does determine which data field you access, but it's not a bijection.

Good news is that in debug mode zig puts a secret safety tag into untagged unions, so if you access the wrong one, you get a crash with a stack trace telling you which field was active and which one you tried to access instead.

In release modes, the tag will be absent, meaning that each Data object will neither have the wasted bytes for the tag, nor its padding.

If you're looking at the Zig codebase for inspiration, don't make the same mistake I did with the `lhs`, `rhs` thing. Better example is [here](https://github.com/ziglang/zig/blob/08d661fcfdfb0a7b99f8bfbf4912acd21cf72d6f/src/Air.zig#L1070-L1073).